### DISCUSSION TITLE
Relay and Led cannot be Coupled in a Complicated Config

### DISCUSSION THREAD

**Discussion by valdus2** [arendst/Tasmota#20930] [Q&A] [answered]
Everything in config is working OK, except,, and I'm completely confused about it, 
why **Led 1 is not coupled with Relay 1.**

You can see from my config below, I have a relay 1, switch 1 and led 1.
The switch controls the relay, but the led never turns on when the relay turns on.

![config](https://github.com/arendst/Tasmota/assets/150083707/f8cc129d-5cfa-4252-8224-0d27fb3ef8a8)

In web GUI toggle 1 turns relay on and off, and toggle 2 turns the display on and off

![ui](https://github.com/arendst/Tasmota/assets/150083707/52873b31-ecf0-4328-971d-1287388a7a87)

The only command I have used is "Ledstate 4" (which works as it should in ledlink), but even changing it doesn't do anything to relay led

I have tried configuring it as Led_i 1 it's always ON whether the relay is ON or OFF

Configuring led gpio as relay_2 i can toggle it on or off, so the led and the gpio are working OK.

(The display I use is an ST7789 with universal driver)

In Web log level 4 this is only what I get:

```
23:17:54.640 SRC: Switch
23:17:54.649 MQT: stat/tasmota_DE6774/RESULT = {"POWER1":"ON"}
23:17:54.653 MQT: stat/tasmota_DE6774/POWER1 = ON
23:17:55.306 CFG: Saved, Count 305, Bytes 4096


23:18:57.019 SRC: WebGui from 171.171.0.0
23:18:57.029 MQT: stat/tasmota_DE6774/RESULT = {"POWER1":"OFF"}
23:18:57.032 MQT: stat/tasmota_DE6774/POWER1 = OFF
23:18:57.219 CFG: Saved, Count 306, Bytes 4096
```

I realy don't know what to do, and I need help desperate.

**Comment by sfromis** [comment]
The index numbers in the dropdown has nothing to do with coupling, only for sequence, always starting at 1, within the same type.

As the [docs about status leds](https://tasmota.github.io/docs/Lights/#status-leds) says
>  If more than one LED component is defined, `Led1`/`Led1i` will act as the Wi-Fi/MQTT status LED and the next defined LED (e.g., `Led2`/`Led2i`) will act as the LED that indicates the power state of the relay(s).

Seems that you can "take over" the Wi-Fi/MQTT status function by configuring a LedLink component, possibly on an unused gpio.

If you want direct control over leds, without any Tasmota built-in behavior around status, you can also config them as relays, and use Berry code or rules to manage when you want them on/off, likely based on the Power channels you want to represent.

Beware of what `LedState` you have, too.

**Comment by valdus2** [comment]
Thanks for the answer, but this is what I have done as you can see from the config and as I described it.

I have defined gpio 4 as LedLink (external led) and it's working as it should (Ledstate 4).

Now, what do i have to do, so, when relay 1 is on, the corresponding led 1 will automatically become on?

I've already use such a config in another project and as long as it concerns relay, led and ledlink everything is working as it should...
Opening the relay, the led comes ON.. You can see the other config below: 

![config2](https://github.com/arendst/Tasmota/assets/150083707/b191f9f0-8da7-4ab5-ab13-d8d12bc91e16)

**Reply by sfromis** [reply]
Did you check what `LedState` you have?

**Comment by valdus2** [comment]
Yes, as I already said LedState 4 and the LedLink led works Ok and it blinks when it publishes MQTT messages.

```
02:16:17.592 CMD: ledstate
02:16:17.594 SRC: WebConsole from 0.0.0.0
02:16:17.597 CMD: Grp 0, Cmd 'LEDSTATE', Idx 1, Len 0, Pld -99, Data ''
02:16:17.603 MQT: stat/tasmota_DE6774/RESULT = {"LedState":4}
```

**Reply by sfromis** [reply, marked as answer]
Well, `LedState 4` does **not** include "show power state". Maybe try with 1 instead.

**Comment by valdus2** [comment]
Yes it works with ledstate 1 ! But now, I don't have the ledlink led blinking when it publishes MQTT messages...
In my other project, it works with LedState 4....So what is going on?

**Reply by valdus2** [reply]
I tried with LedState 5, and now everything is working as my other project which uses LedState 4....

Really confused....Is there any difference between esp32 and esp8266? 
ESP32 devkit : LedState 5
Wemos D1 mini : LedState 4

**Reply by valdus2** [reply]
Nevermind, in my disturbance I was pretty sure about ledstate 4 but actually i had it in ledstate 5.

Considered resolved, and a million thank you again sfromis for enlightening me.

**Reply by valdus2** [reply]
OK, I feel like an idiot...I don't know how to close the discussion as answered...
